lib/Serialization/Deserialization.cpp

```diff
@@ -3254,6 +3254,11 @@ class DeclDeserializer {
     auto namingDecl = cast<ValueDecl>(MF.getDecl(namingDeclID));
     opaqueDecl->setNamingDecl(namingDecl);
 
+    if (auto accessLevel = getActualAccessLevel(rawAccessLevel))
+      opaqueDecl->setAccess(*accessLevel);
```

Comment:
Is the formal access not the same as the owning declaration? You probably don't need to serialize it

Reply:
It should be always the same. But when copying the access level from the owning decl, an assertion was hit due to the reset of the access level. So I've changed the mechanism to always serialize them instead.

Reply:
In that case, you could find all the places we explicitly set the access level of an OpaqueTypeDecl and nuke them, and instead, centralize the logic in FormalAccessRequest::evaluate().

Reply:
The access level assertion isn't in OpaqueTypeDecl but in the naming ValueDecl (a var decl). The correlation is deeper than nuking all sets of OpaqueTypeDecl. I'm not sure the change worths the benefit.